spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/TargetTypeMessageConverter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.converter;
+
+import org.springframework.util.MimeType;
+
+/**
+ * A {@link org.springframework.messaging.converter.MessageConverter} that supports a
+ * target {@link MimeType}/{@link Class} combination.
```

Comment:
this javadoc makes it sound as though there is a single mime-type/class combination that is supported, but according to the interface methods, the contract is such that an arbitrary number of mime-types could be supported, as well as an arbitrary number of target *classes* without any relationship between them (no "combination"... i.e. it doesn't appear to be a 1:1 mapping)